File: client-clojure/src/sirqul_io_t_platform/specs/payment_transaction_response.clj
(ns sirqul-io-t-platform.specs.payment-transaction-response
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [sirqul-io-t-platform.specs.name-string-value-response :refer :all]
            [sirqul-io-t-platform.specs.account-short-response :refer :all]
            [sirqul-io-t-platform.specs.offer-transaction-response :refer :all]
            [sirqul-io-t-platform.specs.purchase-order-item-response :refer :all]
            )
  (:import (java.io File)))


(def payment-transaction-response-data
  {
   (ds/opt :valid) boolean?
   (ds/opt :message) string?
   (ds/opt :version) float?
   (ds/opt :serializeNulls) boolean?
   (ds/opt :startTimeLog) int?
   (ds/opt :errorCode) string?
   (ds/opt :request) (s/coll-of name-string-value-response-spec)
   (ds/opt :paymentTransactionId) int?
   (ds/opt :customer) account-short-response-spec
   (ds/opt :promoCode) string?
   (ds/opt :remoteRefId) string?
   (ds/opt :currencyType) string?
   (ds/opt :amount) float?
   (ds/opt :tax) float?
   (ds/opt :offerTransactions) (s/coll-of offer-transaction-response-spec)
   (ds/opt :orderItems) (s/coll-of purchase-order-item-response-spec)
   (ds/opt :externalPaymentId) string?
   (ds/opt :metadata) string?
   (ds/opt :created) int?
   (ds/opt :updated) int?
   (ds/opt :returning) string?
   })

(def payment-transaction-response-spec
  (ds/spec
    {:name ::payment-transaction-response
     :spec payment-transaction-response-data}))

File: client-scalaz/src/main/scala/org/openapitools/client/api/PaymentTransactionResponse.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import PaymentTransactionResponse._

case class PaymentTransactionResponse (
  valid: Option[Boolean],
message: Option[String],
version: Option[Double],
serializeNulls: Option[Boolean],
startTimeLog: Option[Long],
errorCode: Option[String],
request: Option[List[NameStringValueResponse]],
paymentTransactionId: Option[Long],
customer: Option[AccountShortResponse],
promoCode: Option[String],
remoteRefId: Option[String],
currencyType: Option[String],
amount: Option[Double],
tax: Option[Double],
offerTransactions: Option[List[OfferTransactionResponse]],
orderItems: Option[List[PurchaseOrderItemResponse]],
externalPaymentId: Option[String],
metadata: Option[String],
created: Option[Long],
updated: Option[Long],
returning: Option[String])

object PaymentTransactionResponse {
  import DateTimeCodecs._

  implicit val PaymentTransactionResponseCodecJson: CodecJson[PaymentTransactionResponse] = CodecJson.derive[PaymentTransactionResponse]
  implicit val PaymentTransactionResponseDecoder: EntityDecoder[PaymentTransactionResponse] = jsonOf[PaymentTransactionResponse]
  implicit val PaymentTransactionResponseEncoder: EntityEncoder[PaymentTransactionResponse] = jsonEncoderOf[PaymentTransactionResponse]
}
